tools/isobuild/package-source.js

```diff
@@ -1386,10 +1391,14 @@ _.extend(PackageSource.prototype, {
     // Ignore the usual ignorable files.
     sourceReadOptions.exclude.push(...ignoreFiles);
 
-    // Unless we're running tests, ignore source files with name
-    // "*test*.*", "*tests*.*", "test.*", "tests.*"
+    // Unless we're running tests, ignore all test filenames and if we are, ignore the
+    // type of file we *aren't* running
     if (!global.testCommandMetadata) {
-      Array.prototype.push.apply(sourceReadOptions, TEST_FILENAME_REGEXPS);
+      Array.prototype.push.apply(sourceReadOptions.exclude, TEST_FILENAME_REGEXPS);
```

Comment:
I think you also need to push `UNIT_TEST_FILENAME_REGEXPS` and `INTEGRATION_TEST_FILENAME_REGEXPS` here.

But really, I'd change it a bit and have `TEST_FILENAME_REGEXPS` include both of those already where it's defined (using the ES6 array spread operator), and actually not change the code here.

And then I'd rename `TEST_FILENAME_REGEXPS` to `ALL_TEST_FILENAME_REGEXPS`. That makes the logical relationship between these three arrays match their names.